File: Mini Project/nbc/src/main/java/com/classroom/nbc/service/CustomerService.java
package com.classroom.nbc.service;

import com.classroom.nbc.dto.response.CustomerResponse;
import com.classroom.nbc.dto.response.LoanSummaryResponse;

public interface CustomerService {

    CustomerResponse getProfile(Long customerId);

    LoanSummaryResponse getLoanSummary(Long customerId);
}

File: Mini Project/nbc/frontend/src/app/core/services/customer.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { API_BASE } from '../api.config';
import { CustomerResponse, LoanSummaryResponse } from '../models';

@Injectable({ providedIn: 'root' })
export class CustomerService {
  private readonly http = inject(HttpClient);
  private readonly api = inject(API_BASE);

  getProfile(): Observable<CustomerResponse> {
    return this.http.get<CustomerResponse>(`${this.api}/customers/profile`);
  }

  getLoanSummary(): Observable<LoanSummaryResponse> {
    return this.http.get<LoanSummaryResponse>(`${this.api}/customers/loan-summary`);
  }
}
